playbooks/common/openshift-cluster/initialize_openshift_version.yml

```diff
@@ -0,0 +1,32 @@
+---
+# NOTE: requires openshift_facts be run
+- name: Determine openshift_version to configure on first master
+  hosts: oo_first_master
+  roles:
+  # Using the CLI role here to install the CLI tool/wrapper, and set the
+  # openshift.common.version fact which other hosts can then reference.
+  - openshift_cli
+  pre_tasks:
+  - debug: var=openshift.common.version
+  - debug: var=openshift_version
+  post_tasks:
+  - debug: var=openshift.common.version
+  - debug: var=openshift_version
+
+# NOTE: We set this even on etcd hosts as they may also later run as masters,
+# and we don't want to install wrong version of docker and have to downgrade
+# later.
+- name: Set openshift_version for all hosts
+  hosts: oo_all_hosts
+  tasks:
+  - debug: var=hostvars[groups.oo_first_master.0].openshift.common.version
+  - debug: var=openshift.common.version
+  - debug: var=openshift_version
+    # TODO: Should we use the first master's "openshift_version" var instead of a fact? Could go to just openshift_docker role above, and skip CLI config this early.
```

Comment:
It should be possible to use the openshift_version from the first master, and avoiding set_fact if possible is always nice. One catch is that you'll need to pass in the data from the playbook, rather than trying to introspect the data from the group membership within the roles.